src/DotQuery.Core/AsyncQueryExecutor.cs

```diff
@@ -42,6 +42,23 @@ public Task<TResult> QueryAsync(TQuery query)
         }
 
         /// <summary>
+        /// Execute the given query as an asynchronous operation with a specified cache policy.
+        /// </summary>
+        /// <param name="query">The query to be executed</param>
+        /// <param name="options">The cache policy options.</param>
+        /// <returns>The task object representing the asynchronous operation</returns>
+        public Task<TResult> QueryAsync(TQuery query, CacheEntryOptions options)
```

Comment:
It's a pity that `QueryOptions` is a flags `enum` type, not a `class`. So create a new method signature with `QueryAsync(TQuery, CacheEntryOptions)` for the  the backward compatibility.